.github/workflows/update-cts-json.yaml

```diff
@@ -0,0 +1,46 @@
+name: Update 'cts.json'
+on:
+  push:
+    branches:
+      - main
```

Comment:
I think it'd be good to run the first two steps at the PR level but just not do the commit.  You can prevent that step from running with an `if:` condition that checks to see if we're on `main`.  ([example](https://github.com/gregsdennis/json-everything/blob/master/.github/workflows/dotnet-core.yml#L83))

Reply:
By first two steps you mean "Run build" and "Verify no unexpected changes", right?

Should that maybe be a separate job or workflow then (would complicate the setup though), because for pull requests it should probably not have `contents: write` permission. However, the [documentation](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#how-the-permissions-are-calculated-for-a-workflow-job) says 'write' permissions are changed to 'read' for pull requests from forks.

Reply:
> By first two steps you mean "Run build" and "Verify no unexpected changes", right?

Yes.  These should run as part of the PR.

> Should that maybe be a separate job or workflow then

I find it easier to just include as one workflow.  For example, the workflow I linked to runs building a test badge during a PR to make sure that it works.  But for `master` it also pushes that to a gist that I can then reference in my readme.

Reply:
I have updated the workflow now; I hope that is what you had in mind.